packages/back-end/src/models/BaseModel.ts

```diff
@@ -198,7 +198,11 @@ export abstract class BaseModel<
     // Do nothing by default
   }
   protected async beforeCreate(doc: z.infer<T>, writeOptions?: WriteOptions) {
-    // Do nothing by default
+    if (this.useConfigFile()) {
+      throw new Error(
+        "Cannot create. Segments are being managed by config.yml"
+      );
+    }
```

Comment:
This should live outside `beforeCreate`.  Otherwise, when child classes override this method to add their own logic, they might forget to call `super()` and would unintentionally stop this check from running.

The same places in the code that call `beforeCreate` , `beforeUpdate`, and `beforeDelete` should also do this check.